### PR TITLE
init database

### PR DESCRIPTION
**Pull request by nguyenngocha**

### CODE REVIEW

File: app/models/lesson.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Lesson < ActiveRecord::Base
+  belongs_to :category
+  belongs_to :user
+
+  has_many :user_answers, dependent: :destroy
```
Comment (hellovietnam93) [inline app/models/lesson.rb]: has_many :words, through: :user_answers

File: app/models/relationship.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Relationship < ActiveRecord::Base
+  belongs_to :follower, class_name: "User"
```
Comment (hellovietnam93) [inline app/models/relationship.rb]: User.name

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  has_many :activities, dependent: :destroy
+  has_many :lessons, dependent: :destroy
+  has_many :user_answers, dependent: :destroy
+  has_many :active_relationships, class_name: "Relationship",
+                                  foreign_key: "follower_id",
```
Comment (hellovietnam93) [inline app/models/user.rb]: ident, one tab with has_many

File: app/models/word.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Word < ActiveRecord::Base
+  belongs_to :category
+
+  has_many :answers, dependent: :destroy
+  has_many :user_answers, dependent: :destroy
```
Comment (hellovietnam93) [inline app/models/word.rb]: should add more relation with lessons

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+class User < ActiveRecord::Base
+  has_many :activities, dependent: :destroy
+  has_many :lessons, dependent: :destroy
+  has_many :user_answers, dependent: :destroy
+  has_many :categories, through: :lessons
+  has_many :active_relationships, class_name: "Relationship",
+                                  foreign_key: "follower_id",
```
Comment (hellovietnam93) [inline app/models/user.rb]: lùi so với thằng has_many thôi

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  has_many :activities, dependent: :destroy
+  has_many :lessons, dependent: :destroy
+  has_many :user_answers, dependent: :destroy
+  has_many :categories, through: :lessons
+  has_many :active_relationships,  class_name:  "Relationship",
+                                   foreign_key: "follower_id",
```
Comment (vietmt) [inline app/models/user.rb]: indent

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  has_many :activities, dependent: :destroy
+  has_many :lessons, dependent: :destroy
+  has_many :user_answers, dependent: :destroy
+  has_many :categories, through: :lessons
+  has_many :active_relationships,  class_name:  "Relationship",
```
Comment (vietmt) [inline app/models/user.rb]: class_name:  "Relationship" -> Relationship.class.name

File: app/models/user.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+                                   foreign_key: "follower_id",
+                                   dependent:   :destroy
+  has_many :passive_relationships, class_name:  "Relationship",
+                                   foreign_key: "followed_id",
+                                   dependent:   :destroy
+  has_many :following, through: :active_relationships,  source: :followed
```
Comment (vietmt) [inline app/models/user.rb]: indent

File: app/models/lesson.rb (hunk 1 above)
Comment (nguyenngocha) [inline app/models/lesson.rb]: fixed

File: app/models/word.rb (hunk 4 above)
Comment (nguyenngocha) [inline app/models/word.rb]: fixed

**Pull request merged**